doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/index.md

```diff
@@ -2,304 +2,198 @@
-       $ sudo apt-get update
-       $ sudo apt-get install g++-4.8
-       $ sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.8 50
-       ~~~
 
-2. Install CMake utility:
-    1. For automatic installation, execute the following commands (tested on Ubuntu 14.04):
+        - my-kaa-application
+           - kaa
+           - src
 
-       ~~~ shell
-       $ sudo apt-get install cmake
-       ~~~
+2. Generate C SDK and unpack it to my-kaa-application/kaa directory:
 
-    2. For manual installation, refer to the following example:
+        tar -xvf c-sdk-archive.tar.gz -C my-kaa-application/kaa
 
-       ~~~ shell
-       $ wget http://www.cmake.org/files/v3.3/cmake-3.3.0-rc2.tar.gz
-       $ tar -zxf cmake-3.3.0-rc2.tar.gz
-       $ cd cmake-3.3.0-rc2/
-       $ ./configure
-       $ sudo make install
-       ~~~
+3. Create the application code. Create kaa-application.c file in my-kaa-application/src directory:
 
-3. Install the [**Boost** ](http://www.boost.org/users/download/)libraries:
-    1. For automatic installation, execute the following commands:
+        touch src/kaa-application.c
 
-       ~~~ shell
-       $ sudo apt-get install libboost1.55-all-dev
-       ~~~
+4. Open kaa-application.c and write the application code:
 
-    2. For manual installation, refer to the following example:
 
-       ~~~
-       $ sudo apt-get install libbz2-dev libbz2-1.0 zlib1g zlib1g-dev
-       $ wget http://sourceforge.net/projects/boost/files/boost/1.58.0/boost_1_58_0.tar.gz
-       $ tar -zxf boost_1_58_0.tar.gz
-       $ cd boost_1_58_0/
-       $ ./bootstrap.sh
-       $ sudo ./b2 install
-       ~~~
+        #include <stdio.h>
+        #include <kaa/kaa.h>
+        #include <kaa/platform/kaa_client.h>
+        #include <kaa/kaa_error.h>
 
-4. Install the [**AvroC++**](http://avro.apache.org/docs/1.7.6/api/cpp/html/index.html) library manually:
 
-   ~~~
-   $ wget http://archive.apache.org/dist/avro/avro-1.7.5/cpp/avro-cpp-1.7.5.tar.gz
-   $ tar -zxf avro-cpp-1.7.7.tar.gz
-   $ cd avro-cpp-1.7.5/
-   $ cmake -G "Unix Makefiles"
-   $ sudo make install
-   ~~~
+        #define KAA_APP_PRINT_TIMES 5
 
-5. Install the [**Botan**](http://botan.randombit.net/) library by executing the following command:
+        static void dummy_function(void* context)
+        {
+            for (size_t i=1; i<KAA_APP_PRINT_TIMES; i++) {
+                printf("Hello, I am a Kaa Application!\n");
+            }
+            kaa_client_stop(context);
+        }
 
-   ~~~
-   $ wget https://github.com/randombit/botan/archive/1.11.28.tar.gz
-   $ tar -zxf 1.11.28.tar.gz
-   $ cd botan-1.11.28/
-   $ ./configure.py
-   $ sudo make install
-   $ sudo ln -s /usr/local/include/botan-1.11/botan /usr/local/include/botan
-   ~~~
+        int main()
+        {
+            kaa_client_t* kaa_client = NULL;
+            kaa_error_t error = kaa_client_create(&kaa_client, NULL);
+            if (error) {
+                return -1;
```

Comment:
Domain of the return values is implementation-defined. This means that environment can interpret it as unsigned integer. Use http://en.cppreference.com/w/c/program/EXIT_status